### PR TITLE
Removed last usage of the internal container

### LINKED ISSUE #448: Replace the internal container with newing up instances directly where needed
@SeanFeldman and I made a spike to show that it is possible to replace the internal container inside the topology. The approach would be following

- The topology is the composition root of the infrastructure required to run it
- All singleton scoped objects are held as fields inside the topology
- On demand created objects with transient scope are created with `new `when needed and held by the instance that needs the transient objects
- All container resolve calls happening during runtime are replaced with constructor injection or explicit method injection (method accepting parameters which define what dependencies an instance needs).

### PR DESCRIPTION
**Pull request by danielmarbach**
Closes #448

### CODE REVIEW

File: src/Transport/Topology/Topologies/EndpointOrientedTopology.cs [hunk 1]
```diff
@@ -24,23 +24,17 @@ class EndpointOrientedTopologyInternal : ITopologyInternal
...
-
-        internal EndpointOrientedTopologyInternal(ITransportPartsContainerInternal container)
-        {
-            this.container = container;
-        }
-
         public bool HasNativePubSubSupport => true;
         public bool HasSupportForCentralizedPubSub => true;
         public TopologySettings Settings { get; } = new TopologySettings();
+        public ITopologySectionManagerInternal TopologySectionManager => topologySectionManager;
```
Comment (yvesgoeleven) [inline src/Transport/Topology/Topologies/EndpointOrientedTopology.cs]: can't this be internal? It's only exposed for testing purposes

File: src/Transport/Topology/Topologies/EndpointOrientedTopology.cs [hunk 2]
```diff
@@ -24,23 +24,17 @@ class EndpointOrientedTopologyInternal : ITopologyInternal
...
+        public IOperateTopologyInternal Operator => topologyOperator;
```
Comment (yvesgoeleven) [inline src/Transport/Topology/Topologies/EndpointOrientedTopology.cs]: can't this be internal? It's only exposed for testing purposes

File: src/Transport/Topology/Topologies/ForwardingTopology.cs [hunk 3]
```diff
@@ -31,17 +30,11 @@ class ForwardingTopologyInternal : ITopologyInternal
...
-        internal ForwardingTopologyInternal(ITransportPartsContainerInternal container)
-        {
-            this.container = container;
-        }
-
         public bool HasNativePubSubSupport => true;
         public bool HasSupportForCentralizedPubSub => true;
         public TopologySettings Settings { get; } = new TopologySettings();
-
+        public ITopologySectionManagerInternal TopologySectionManager => topologySectionManager;
```
Comment (yvesgoeleven) [inline src/Transport/Topology/Topologies/ForwardingTopology.cs]: can't this be internal? It's only exposed for testing purposes

File: src/Transport/Topology/Topologies/ForwardingTopology.cs [hunk 4]
```diff
@@ -31,17 +30,11 @@ class ForwardingTopologyInternal : ITopologyInternal
...
+        public IOperateTopologyInternal Operator => topologyOperator;
```
Comment (yvesgoeleven) [inline src/Transport/Topology/Topologies/ForwardingTopology.cs]: can't this be internal? It's only exposed for testing purposes

File: src/Transport/Topology/Topologies/EndpointOrientedTopology.cs (hunk 1 above)
Comment (yvesgoeleven) [inline src/Transport/Topology/Topologies/EndpointOrientedTopology.cs]: class itself is internal

File: src/Transport/Topology/Topologies/EndpointOrientedTopology.cs (hunk 2 above)
Comment (yvesgoeleven) [inline src/Transport/Topology/Topologies/EndpointOrientedTopology.cs]: class itself is internal

File: src/Transport/Topology/Topologies/ForwardingTopology.cs (hunk 3 above)
Comment (yvesgoeleven) [inline src/Transport/Topology/Topologies/ForwardingTopology.cs]: class itself is internal

File: src/Transport/Topology/Topologies/ForwardingTopology.cs (hunk 4 above)
Comment (yvesgoeleven) [inline src/Transport/Topology/Topologies/ForwardingTopology.cs]: class itself is internal

**Comment by yvesgoeleven** [comment]
Build server says a few inspections need to be fixed, but besides that, good to go for me

**Comment by danielmarbach** [comment]
@yvesgoeleven green now

**Pull request merged**